pkg/security/probe/probe.go

```diff
@@ -1233,21 +1232,32 @@ func (p *Probe) GetDebugStats() map[string]interface{} {
 	return debug
 }
 
-// NewRuleSet returns a new rule set
-func (p *Probe) NewRuleSet(eventTypeEnabled map[eval.EventType]bool) *rules.RuleSet {
-	ruleOpts, evalOpts := rules.NewEvalOpts(eventTypeEnabled)
+// NewEvaluationSet returns a new evaluation set with rule sets tagged by the passed-in tag values for the "ruleset" tag key
+func (p *Probe) NewEvaluationSet(eventTypeEnabled map[eval.EventType]bool, ruleSetTagValues []string) (*rules.EvaluationSet, error) {
+	var ruleSetsToInclude []*rules.RuleSet
+	for _, ruleSetTagValue := range ruleSetTagValues {
+		ruleOpts, evalOpts := rules.NewEvalOpts(eventTypeEnabled)
 
-	ruleOpts.WithLogger(seclog.DefaultLogger)
-	ruleOpts.WithSupportedDiscarders(SupportedDiscarders)
-	ruleOpts.WithReservedRuleIDs(events.AllCustomRuleIDs())
+		ruleOpts.WithLogger(seclog.DefaultLogger)
+		ruleOpts.WithSupportedDiscarders(SupportedDiscarders)
```

Comment:
The problem here is that we instantiate all the ruleset with the same options. The threat score ruleset doesn't require discarders.